internal/experiment/webconnectivity/control.go

```diff
@@ -47,10 +46,10 @@ func Control(
 //
 // This is very useful to know what ASNs were the IP addresses returned by
 // the control according to the probe's ASN database.
-func fillASNs(dns *ControlDNSResult) {
+func fillASNs(sess model.ExperimentSession, dns *ControlDNSResult) {
 	dns.ASNs = []int64{}
 	for _, ip := range dns.Addrs {
-		asn, _, _ := geoipx.LookupASN(ip)
+		asn, _, _ := sess.LookupASN(ip)
```

Comment:
This change is not enough. We also need to override how we compute the ASN for probe's own lookups, otherwise there is the high risk we will always see inconsistencies.